tool/common/labels.go

```diff
@@ -0,0 +1,86 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/client"
+)
+
+// LabelFilter is a common type for parsing Teleport labels from commandline args
+// using kingpin
```

Comment:
Why do we need these new types? We have lots of --labels flags with kingpin and haven't needed this to date.

Reply:
I was hoping to start using this as the one canonical implementation of CLI label parsing, and migrate the other ad-hoc implementations to use this over time. 

I can see it as a possible "and now you have _15_ standards", though.

Reply:
Not a bad idea, but I'd do that as a separate PR. If we merge this as-is I'd wager there's a > 50% chance that we never get to updating the existing usages and then we have multiple ways to parse label flags rather than being consistent.